pkg/server/backend_manager.go

```diff
@@ -70,24 +71,15 @@ func GenProxyStrategiesFromStr(proxyStrategies string) ([]ProxyStrategy, error)
 	return ps, nil
 }
 
-type Backend interface {
-	Send(p *client.Packet) error
-	Recv() (*client.Packet, error)
-	Context() context.Context
-}
-
-var _ Backend = &backend{}
-var _ Backend = agent.AgentService_ConnectServer(nil)
-
-type backend struct {
-	// TODO: this is a multi-writer single-reader pattern, it's tricky to
-	// write it using channel. Let's worry about performance later.
+// Backend wraps an agent.AgentService_ConnectServer, provides
+// synchronization and emits common stream metrics.
+type Backend struct {
```

Comment:
Mixed opinions about removing the Backend abstraction. With the interface we were in a place to be able to swap out the communication system between Agent and Server. With this change we are dictating it must be our GRPC tunnel. The original idea that if a provider wanted to swap it out for something like a VPN tunnel it would be relatively easy to do so. Granted that use case has never some up but wondering if we really want to lose that.